src/modules/permissionhooks/AllowlistModule.sol

```diff
@@ -25,48 +21,53 @@ contract AllowlistModule is IValidationHook, BaseModule {
         bool hasSelectorAllowlist;
     }
 
-    mapping(address target => mapping(address account => AllowlistEntry)) public targetAllowlist;
-    mapping(address target => mapping(bytes4 selector => mapping(address account => bool))) public
-        selectorAllowlist;
+    mapping(uint32 entityId => mapping(address target => mapping(address account => AllowlistEntry))) public
+        targetAllowlist;
+    mapping(
+        uint32 entityId => mapping(address target => mapping(bytes4 selector => mapping(address account => bool)))
+    ) public selectorAllowlist;
```

Comment:
We will need some events when those two mappings are updated. Otherwise, there is no easy way to get those info for an account (especially if more than more apps add hooks).

Reply:
Do we definitely need the events? It's relatively easy to detect "is this call allowed?" by just reading the mappings directly, and if it's not, the wallet / SDK can generate a call to update the permissions. I'm not sure we would need to generate a history of the permission changes that events would give us.

Might we worth making the `_checkAllowlistCalldata` internal function public, to reduce the number of round trips a wallet / SDK needs to check if a call is allowed.

Reply:
Each Dapp might not need it for now with the assumption on they store those information themselves. But the wallet or whoever wants to show users their account info will need to access the whole list. Without that, the user will have a hard time to get a full state of the account.